core/contracts/financial_templates/Liquidatable.sol

```diff
@@ -0,0 +1,392 @@
+    // Represented as a multipler, see above
+    FixedPoint.Unsigned disputerDisputeRewardPct;
+
+    /**
+     * Method modifiers
+     */
+
+    // Callable before the liquidation's expiry AND there is no pending dispute on the liquidation
+    modifier onlyPreExpiryAndPreDispute(uint id, address sponsor) {
+        require(
+            (getCurrentTime() < _getLiquidationData(sponsor, id).expiry) &&
+                (_getLiquidationData(sponsor, id).state == Status.PreDispute),
+            "Liquidation has expired or has already been disputed"
+        );
+        _;
+    }
+    // Callable either post the liquidation's expiry or after a dispute has been resolved,
+    // i.e. once a dispute has been requested, the liquidation's expiry ceases to matter
+    modifier onlyPostExpiryOrPostDispute(uint id, address sponsor) {
+        require(
+            (_getLiquidationData(sponsor, id).state == Status.DisputeSucceeded) ||
+                (_getLiquidationData(sponsor, id).state == Status.DisputeFailed) ||
+                ((_getLiquidationData(sponsor, id).expiry <= getCurrentTime()) &&
+                    (_getLiquidationData(sponsor, id).state == Status.PreDispute)),
+            "Liquidation has not expired or is pending dispute"
+        );
+        _;
+    }
+    // Callable only after a liquidation has been disputed but has not yet resolved
+    modifier onlyPendingDispute(uint id, address sponsor) {
+        require(
+            _getLiquidationData(sponsor, id).state == Status.PendingDispute,
+            "Liquidation is not currently pending dispute"
+        );
+        _;
+    }
+
+    /**
+     * Constructor: set universal Liquidation variables
+     */
+    constructor(
+        bool _isTest,
+        uint _positionExpiry,
+        uint _positionWithdrawalLiveness,
+        address _collateralCurrency,
+        FixedPoint.Unsigned memory _disputeBondPct,
+        FixedPoint.Unsigned memory _sponsorDisputeRewardPct,
+        FixedPoint.Unsigned memory _disputerDisputeRewardPct,
+        uint _liquidationLiveness
+    ) public PricelessPositionManager(_positionExpiry, _positionWithdrawalLiveness, _collateralCurrency, _isTest) {
+        require(
+            _sponsorDisputeRewardPct.add(_disputerDisputeRewardPct).isLessThan(1),
+            "Sponsor and Disputer dispute rewards shouldn't sum to 100% or more"
+        );
+        disputeBondPct = _disputeBondPct;
+        sponsorDisputeRewardPct = _sponsorDisputeRewardPct;
+        disputerDisputeRewardPct = _disputerDisputeRewardPct;
+        liquidationLiveness = _liquidationLiveness;
+    }
+
+    /**
+     * Liquidate's the sponsor's position if the caller has enough
+     * synthetic tokens to retire the position's outstanding tokens.
+     *
+     * This method will generate an ID that will uniquely identify liquidation
+     * for the sponsor.
+     * TODO: Perhaps pass this ID via an event rather than a return value
+     *
+     * TODO: Possibly allow partial liquidations
```

Comment:
As discussed today, partial liquidations will be easy to add given current implementation but might introduce nasty attack vectors. For example, sending a bunch of tiny partial liquidations instead of one "full" liquidation